scripts/rails_console/deduplicate_question_uids.rb

```diff
@@ -0,0 +1,397 @@
+# this script resolves an issue where some questions share uids that are identical besides their case, leading to wonky behavior in internal tools
+
+# RUN THIS IN THE LMS RAILS CONSOLE
+
+uids = Question.all.pluck(:uid)
+normalized_groups = Hash.new { |hash, key| hash[key] = [] }
+
+# group UIDs by their lowercase version
+uids.each do |uid|
+  normalized_groups[uid.downcase] << uid
+end
+
+# select only the groups with more than one element
+duplicate_groups = normalized_groups.values.select { |group| group.size > 1 }
+
+# RUN THIS IN THE CMS RAILS console
+
+# COPY AND PASTE IN DUPLICATE GROUPS
+
+# this took about ten minutes on the production CMS rails console
+duplicate_groups_with_counts = duplicate_groups.map do |duplicate_group|
+  duplicate_group.map do |uid|
+    { uid: uid, count: Response.where(question_uid: uid).size }
+  end
+end
+
+lower_count_questions = duplicate_groups_with_counts.filter { |dg| dg.length > 1 }.map do |duplicate_group|
+  question_to_change = duplicate_group[0][:count] > duplicate_group[1][:count] ? duplicate_group[1] : duplicate_group[0]
+  question_to_change[:new_uid] = SecureRandom.uuid
+  question_to_change
+end
+
+# COPY LOWER COUNT QUESTIONS SOMEWHERE AND SAVE IT
+
+=begin
+lower_count_questions = [{:uid=>"-Jzw0qjJeHLqN6eDQTJk", :count=>349, :new_uid=>"66586fd5-30c5-4359-9ba8-cbfba8b3ec16"},
```

Comment:
Take or leave, since this may not be worth the effort (I'm on the fence about it). You could consider hosting this as a csv file and downloading parsing the CSV so you don't have to copy/paste this data in multiple places. Even if the file is just in github, you can download the raw version as a csv. Anyway, take or leave.

```ruby
require 'open-uri'
require 'csv'

csv_data = CSV.readlines(open(download_url), headers: true)